File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/BackupClientLink.scala
package io.swagger.server.model


/**
 * A navigation link for backup service actions in the client portal.
 *
 * @param label Label of the client link.
 * @param link Link URL of the client link.
 * @param icon Icon of the client link.
 * @param icon_text Icon text of the client link.
 * @param help_text Help text of the client link.
 * @param other_attr Other attributes of the client link.
 */
case class BackupClientLink (
  label: Option[String],
  link: Option[String],
  icon: Option[String],
  icon_text: Option[String],
  help_text: Option[String],
  other_attr: Option[String]
)

File: OPENAPI-REST-API/openapi-client/php-dt/src/App/DTO/BackupClientLink.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\PhpAttribute as DTA;

/**
 * A navigation link for backup service actions in the client portal.
 */
class BackupClientLink
{
    /**
     * Label of the client link.
     */
    #[DTA\Data(field: "label", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $label = null;

    /**
     * Link URL of the client link.
     */
    #[DTA\Data(field: "link", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $link = null;

    /**
     * Icon of the client link.
     */
    #[DTA\Data(field: "icon", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $icon = null;

    /**
     * Icon text of the client link.
     */
    #[DTA\Data(field: "icon_text", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $icon_text = null;

    /**
     * Help text of the client link.
     */
    #[DTA\Data(field: "help_text", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $help_text = null;

    /**
     * Other attributes of the client link.
     */
    #[DTA\Data(field: "other_attr", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $other_attr = null;

}
